### PR TITLE
Signup Route is not accessible to logged In users

### LINKED ISSUE #637: Signup route is accessible to logged in users
**Actual Behaviour**
When a logged in user visit's https://accounts.susi.ai/signup, It is accessible.

**Expected Behaviour**
Logged In user should not be allowed to visit that route.

**Steps to reproduce it**

1) Visit https://accounts.susi.ai/signup without Logging to your account.

**Would you like to work on the issue?**

Yes, I would like to work on the issue.

### PR DESCRIPTION
**Pull request by plxity**
Fixes #637

Changes: Signup route is now not accessible to logged in users.

Surge Deployment Link: https://pr-690-fossasia-susi-accounts.surge.sh

### CODE REVIEW

**Comment by plxity** [comment]
@AakashMallik I have used life cycle method here to check if the user is logged in or not. Or should I create a component like ProtectedRoute?

File: src/components/Auth/SignUp/SignUp.react.js [hunk 1]
```diff
@@ -28,6 +29,13 @@ import zxcvbn from 'zxcvbn';
 import './SignUp.css';
 
 export default class SignUp extends Component {
+  componentDidMount() {
+    const cookies = new Cookies();
```
Comment (AakashMallik) [inline src/components/Auth/SignUp/SignUp.react.js]: create the cookies outside the class defination. That would look more consistent to the rest of the code base.

File: src/components/Auth/SignUp/SignUp.react.js [hunk 2]
```diff
@@ -28,6 +29,13 @@ import zxcvbn from 'zxcvbn';
...
+    const isLoggedIn = !!cookies.get('loggedIn');
```
Comment (AakashMallik) [inline src/components/Auth/SignUp/SignUp.react.js]: Why **!!** ?

Comment (AakashMallik) [reply src/components/Auth/SignUp/SignUp.react.js]: Also, avoid the extra variable. use it directly in the if statement

Comment (plxity) [reply src/components/Auth/SignUp/SignUp.react.js]: Okay done.

File: src/components/Auth/SignUp/SignUp.react.js [hunk 3]
```diff
@@ -17,17 +17,23 @@ import ChatConstants from '../../../constants/ChatConstants';
...
-
+import Cookies from 'universal-cookie';
 import { urls } from '../../../Utils';
 import { CAPTCHA_KEY } from '../../../config.js';
 import Recaptcha from 'react-recaptcha';
 import zxcvbn from 'zxcvbn';
 
 import './SignUp.css';
-
+const cookies = new Cookies();
```
Comment (AakashMallik) [inline src/components/Auth/SignUp/SignUp.react.js]: import './SignUp.css';

const cookies = new Cookies();

export default class SignUp extends Component

Comment (AakashMallik) [reply src/components/Auth/SignUp/SignUp.react.js]: new lines

**Comment by plxity** [comment]
@akshatnitd @anshumanv please review this.

**Comment by plxity** [comment]
@AdityaSrivast Link working fine. Please check again.

**Comment by AdityaSrivast** [comment]
image

@plxity I'm still able to access the route.

**Comment by plxity** [comment]
@AdityaSrivast I checked it locally again, It is working perfectly fine. Lemme check the surge link once.

**Comment by plxity** [comment]
@akshatnitd @anshumanv please review this.

**Comment by anshumanv** [comment]
Doesn't seem to work for me in the surge link

image

**Comment by plxity** [comment]
@anshumanv Locally it works perfectly fine. I don't know what happens in surge links (sometimes it works). I am trying to figure it out.

**Pull request merged**